darwin/Platform.h

```diff
@@ -34,6 +35,10 @@ extern const SignalItem Platform_signals[];
 
 extern const unsigned int Platform_numberOfSignals;
 
+extern const MemoryClass Platform_memoryClasses[];
+
+extern const unsigned int Platform_numberOfMemoryClasses;
```

Comment:
For better optimization I would like this `Platform_numberOfMemoryClasses` to go away.

The purpose of that global constant can be replaced with `ARRAYSIZE(Platform_memoryClasses)` even though letting the `sizeof` operator work would require exposing the table to global scope.